opm/simulators/wells/MultisegmentWell_impl.hpp

```diff
@@ -1210,14 +1210,10 @@ namespace Opm
         setToZero(wellPI);
 
         const auto preferred_phase = this->well_ecl_.getPreferredPhase();
+        auto subsetPerfID   = 0;
```

Comment:
I'd prefer if you didn't introduce a type-conversion here.  The original expression was crafted to generate a zero of the appropriate type.

Reply:
What is the correct type? It seems like there are two usages below: 
`getMobility(ebosSimulator, static_cast<int>(subsetPerfID), mob);` and fluidState(subsetPerfID)

Reply:
... and both seem to take an `int` and not a `std::size_t`. Shouldn't my changes be at least as correct as before or does auto not result in an `int` here?

Reply:
> both seem to take an `int` and not a `std::size_t`. Shouldn't my changes be at least as correct as before or does auto not result in an `int` here?

Ehm, yes&mdash;you're right.  I was thinking of an earlier version of this code that never made it to GitHub and which used `subsetPerfID` differently.  I'll merge this now.